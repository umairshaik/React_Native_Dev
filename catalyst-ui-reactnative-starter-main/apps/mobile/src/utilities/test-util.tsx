import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {ThemeProvider} from '../context/theme/themeContext';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});
const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <ThemeProvider>
    <QueryClientProvider client={query}>{children}</QueryClientProvider>
  </ThemeProvider>
);

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export * from '@testing-library/react-native';
export {customRender as render};
