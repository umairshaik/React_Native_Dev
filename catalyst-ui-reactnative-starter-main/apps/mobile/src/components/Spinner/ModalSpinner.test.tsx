import {render} from '../../utilities/test-util';
import {ScreenLoader} from './ModalSpinner';

describe('Modal Spinner', () => {
  test('render spinner component', () => {
    const component = render(<ScreenLoader />);
    expect(component).toMatchSnapshot();
  });

  test('render spinner view', () => {
    const component = render(<ScreenLoader />);
    expect(component.getByTestId('global-loader-modal')).toBeTruthy();
  });
});
