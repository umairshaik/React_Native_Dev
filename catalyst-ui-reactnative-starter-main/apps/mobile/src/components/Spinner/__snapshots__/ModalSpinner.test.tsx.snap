// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Spinner render spinner component 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.3)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
    testID="global-loader-modal"
  >
    <ActivityIndicator
      color="black"
      size="large"
    />
  </View>
</Modal>
`;
